microservices/matching-service/src/client.ts

```diff
@@ -0,0 +1,52 @@
+import amqp from 'amqplib';
+import { MatchRequest } from './types';
+
+const QUEUE_NAME = 'matching_service_queue';
+
+const HOST = '127.0.0.1';
+const PORT = '5672';
+
+export async function findMatch(request: MatchRequest) {
+  const connection = await amqp.connect(`amqp://${HOST}:${PORT}`);
+  const channel = await connection.createChannel();
+  const q = await channel.assertQueue('', { exclusive: true });
+
+  const correlationId = generateUuid();
+
+  const consumerPromise = new Promise<string>((resolve) => {
+    channel.consume(
+      q.queue,
+      function(msg) {
+        if (!msg) {
+          console.log('No message found on consume, dropping packet');
+          return;
+        }
+        if (msg.properties.correlationId != correlationId) {
+          console.log('Request id does not match, dropping packet');
+        }
+        connection.close();
+        resolve(msg.content.toString());
+      },
+      {
+        noAck: true
+      }
+    );
+  });
+
+  channel.sendToQueue(
+    QUEUE_NAME,
+    Buffer.from(JSON.stringify(request)),
+    {
+      correlationId: correlationId,
+      replyTo: q.queue
+    }
+  );
+
+  return consumerPromise;
+}
+
+function generateUuid() {
+  return Math.random().toString() +
+         Math.random().toString() +
+         Math.random().toString();
+}
```

Comment:
There are packages for uuid generation -- see [`nanoid`](https://www.npmjs.com/package/nanoid) or [`uuid`](https://www.npmjs.com/package/uuid)

Reply:
Resolved in [6d0c79c](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g37/pull/14/commits/6d0c79c8c7227c5be27d09e9a5485950b72de899)